### ISSUE TITLE
Build fails with gcc 8

### ISSUE THREAD

**Issue by esabol** [gearman/gearmand#213]
```
libgearman-server/plugins/queue/redis/queue.cc: In member function ‘bool gearmand::plugins::queue::Hiredis::hmset(vchar_t, const void*, size_t, uint32_t)’:
libgearman-server/plugins/queue/redis/queue.cc:91:19: error: type qualifiers ignored on cast result type [-Werror=ignored-qualifiers]
     (const size_t)5,
                   ^
libgearman-server/plugins/queue/redis/queue.cc:92:28: error: type qualifiers ignored on cast result type [-Werror=ignored-qualifiers]
     (const size_t)key.size(),
                            ^
libgearman-server/plugins/queue/redis/queue.cc:93:19: error: type qualifiers ignored on cast result type [-Werror=ignored-qualifiers]
     (const size_t)4,
                   ^
libgearman-server/plugins/queue/redis/queue.cc:94:19: error: type qualifiers ignored on cast result type [-Werror=ignored-qualifiers]
     (const size_t)data_size,
                   ^~~~~~~~~
libgearman-server/plugins/queue/redis/queue.cc:95:19: error: type qualifiers ignored on cast result type [-Werror=ignored-qualifiers]
     (const size_t)8,
                   ^
cc1plus: all warnings being treated as errors
```

**Comment by esabol** [comment]
Removing "const" from those casts resolves those errors.

Next up:
```
libtest/http.cc: In function ‘size_t libtest::http::http_get_result_callback(void*, size_t, size_t, void*)’:
libtest/http.cc:90:39: error: ‘void* memcpy(void*, const void*, size_t)’ writing to an object of type ‘libtest::vchar_t’ {aka ‘class std::vector<char>’} with no trivial copy-assignment; use copy-assignment or copy-initialization instead [-Werror=class-memaccess]
   memcpy(&_body[0], ptr, _body->size());
                                       ^
In file included from /usr/include/c++/8/vector:64,
                 from ./libtest/vchar.hpp:43,
                 from ./libtest/test.hpp:67,
                 from ./libtest/common.h:85,
                 from libtest/http.cc:39:
/usr/include/c++/8/bits/stl_vector.h:339:11: note: ‘libtest::vchar_t’ {aka ‘class std::vector<char>’} declared here
     class vector : protected _Vector_base<_Tp, _Alloc>
           ^~~~~~
```

**Comment by esabol** [comment]
Change that `memcpy()` call to the following:
```C
     memcpy(static_cast<void*>(&_body[0]), ptr, _body->size());
```

Next up:
```
tests/httpd_test.cc: In function ‘test_return_t curl_function_TEST(void*)’:
tests/httpd_test.cc:101:44: error: ‘%s’ directive output may be truncated writing 12 bytes into a region of size between 1 and 1024 [-Werror=format-truncation=]
   snprintf(worker_url, sizeof(worker_url), "%s%s", host_url, WORKER_FUNCTION_NAME);
                                            ^~~~~~
In file included from /usr/include/stdio.h:937,
                 from /usr/include/c++/8/cstdio:42,
                 from ./libtest/test.hpp:60,
                 from tests/httpd_test.cc:40:
/usr/include/x86_64-linux-gnu/bits/stdio2.h:64:35: note: ‘__builtin___snprintf_chk’ output between 13 and 1036 bytes into a destination of size 1024
   return __builtin___snprintf_chk (__s, __n, __USE_FORTIFY_LEVEL - 1,
          ~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        __bos (__s), __fmt, __va_arg_pack ());
        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
tests/httpd_test.cc: In function ‘test_return_t curl_function_no_body_TEST(void*)’:
tests/httpd_test.cc:83:44: error: ‘%s’ directive output may be truncated writing 12 bytes into a region of size between 1 and 1024 [-Werror=format-truncation=]
   snprintf(worker_url, sizeof(worker_url), "%s%s", host_url, WORKER_FUNCTION_NAME);
                                            ^~~~~~
In file included from /usr/include/stdio.h:937,
                 from /usr/include/c++/8/cstdio:42,
                 from ./libtest/test.hpp:60,
                 from tests/httpd_test.cc:40:
/usr/include/x86_64-linux-gnu/bits/stdio2.h:64:35: note: ‘__builtin___snprintf_chk’ output between 13 and 1036 bytes into a destination of size 1024
   return __builtin___snprintf_chk (__s, __n, __USE_FORTIFY_LEVEL - 1,
          ~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        __bos (__s), __fmt, __va_arg_pack ());
        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  CXXLD    libtest/libtest.la
  CCLD     libhostile/t/accept
cc1plus: all warnings being treated as errors
```

**Comment by esabol** [comment]
Changing "char worker_url[1024];" to "char worker_url[1024 + sizeof(WORKER_FUNCTION_NAME) - 1];" fixes those.

PR coming as soon I figure out what's wrong with my fork....

**Comment by esabol** [comment]
Spoke too soon. If you `./configure --enable-ssl`, another error appears:
```
libgearman-server/io.cc: In function ‘size_t _connection_read(gearman_server_con_st*, void*, size_t, gearmand_error_t&)’:
libgearman-server/io.cc:153:11: error: this statement may fall through [-Werror=implicit-fallthrough=]
           if (errno) // If errno is really set, then let our normal error logic handle.
           ^~
libgearman-server/io.cc:159:9: note: here
         case SSL_ERROR_SSL:
         ^~~~
cc1plus: all warnings being treated as errors
```

Travis CI builds should compile with `--enable-ssl` for full coverage of the code. Anyone disagree?

**Comment by esabol** [comment]
This fixes that error, of course:
```
diff --git a/libgearman-server/io.cc b/libgearman-server/io.cc
index 2cd6f7d9..48fe24c7 100644
--- a/libgearman-server/io.cc
+++ b/libgearman-server/io.cc
@@ -113,6 +113,9 @@ const char* gearmand_io_st::port() const
   return "-";
 }
 
+#if __GNUC__ >= 7
+  #pragma GCC diagnostic warning "-Wimplicit-fallthrough"
+#endif
 static size_t _connection_read(gearman_server_con_st *con, void *data, size_t data_size, gearmand_error_t &ret)
 {
   ssize_t read_size;
```

**Comment by p-alik** [comment]
fixed in #219
